### PR TITLE
Improved Edit Bird/Userbar

### PR DESCRIPTION
**Pull request by mr-winter**
We built a non-iframe version of the Wagtail userbar, according to the quick mockup we proposed earlier this week. This addresses #2164 and still needs some updates on the tests wagtail.wagtailadmin.tests.test_userbar.TestUserbarTag

### CODE REVIEW

**Comment by tomdyson** [comment]
Thanks @mr-winter! This looks great.

File: wagtail/wagtailadmin/static_src/wagtailadmin/js/userbar.js [hunk 1]
```diff
@@ -1,18 +1,33 @@
 'use strict';
 
 document.addEventListener('DOMContentLoaded', function() {
-    var body = document.querySelectorAll('body')[0];
-    var nav = document.querySelectorAll('nav')[0];
-    var className = 'ready';
-    var hasPostMessage = window.postMessage;
+    var trigger = document.querySelectorAll('[data-wagtail-userbar-trigger]')[0];
```
Comment (akx) [inline wagtail/wagtailadmin/static_src/wagtailadmin/js/userbar.js]: If you're only looking for the first object to match, you can use `.querySelector` instead.

Comment (mr-winter) [inline wagtail/wagtailadmin/static_src/wagtailadmin/js/userbar.js]: True! I fixed that. Thanks

File: wagtail/wagtailadmin/templates/wagtailadmin/userbar/base.html [hunk 2]
```diff
@@ -1,24 +1,19 @@
-{% extends "wagtailadmin/skeleton.html" %}
-{% load staticfiles i18n %}
-
-{% block titletag %}{% trans 'User bar' %}{% endblock %}
-{% block css %}
+{% load compress static i18n %}
```
Comment (kaedroho) [inline wagtail/wagtailadmin/templates/wagtailadmin/userbar/base.html]: ``static`` should be ``staticfiles``. And ``compress`` should be removed

**Comment by JoshBarr** [comment]
I've made a few changes to the CSS + HTML, to ensure all the classNames are prefixed with `.wagtail-`, including all the icon classes. There's some duplication but I'll take your guidance @davecranwell as to whether we change the `_icons.scss` partial to use this sass factory method. 

Namespacing the classNames should help avoid style conflicts with the embedding page. We might want to make the namespace easier to change, and set it something more obscure, like `.wtCMSUserBar-`? 

Also added configurable corners to the userbar: `{% wagtailuserbar 'bottom-right' %}`.

Parameters accepted are:

```
top-left
top-right
bottom-left
bottom-right
```

That's all for now.

**Comment by JoshBarr** [comment]
Gah, still need to fix the lint errors. I swear the CSS isn't that badly formatted :)

**Comment by davecranwell** [comment]
One minor thing we realised while reviewing this today: the new and super-useful "Show in explorer" button will presumably not work if the page in question isn't on the first page of pages? i.e the Explorer paginates 50 page items at a time (iirc) and if the page in question is on pagination page 2, it couldn't be "Shown" as the phrase suggests it might. I wonder if theres's a slightly different version of this phrase we could use instead.

**Comment by JoshBarr** [comment]
Good spotting!

Is there a way to do reverse lookup for paginated items in Django? It would
be great to append the query parameter to the URL so it jumps to the right
page.

If not, we could say something like "show parent in explorer", "go to
parent", "show siblings" or even "explore". Not sure any of those are quite
right.

**Comment by tomdyson** [comment]
> Is there a way to do reverse lookup for paginated items in Django?

Not that I know of, and it sounds pretty fiddly. FWIW I don't think it matters much if the 'show in explorer' button takes you to the first of many pages. Not enough to block this PR, anyway! Great work, @mr-winter and @JoshBarr.

**Comment by gasman** [comment]
I reckon working out the page number of a given item in a queryset is doable, and would make a neat addition to the django.core.paginator API: check what field(s) the queryset is ordered on, do a 'count' query to find the number of items whose value on that field is less than the current item's, and divide by the page size. But yep, definitely out of scope for this, and not a blocker for merging!

**Comment by tomdyson** [comment]
@JoshBarr can you please merge this if you're happy with everything?

**Pull request merged**